src/test/java/org/xstefank/check/LatestCommitCheckTest.java

```diff
@@ -0,0 +1,62 @@
+package org.xstefank.check;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.xstefank.TestUtils;
+import org.xstefank.api.GitHubAPI;
+import org.xstefank.model.yaml.RegexDefinition;
+import java.util.regex.Pattern;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(GitHubAPI.class)
+public class LatestCommitCheckTest {
+
+    private RegexDefinition testCommit = new RegexDefinition();
+    private LatestCommitCheck latestCommitCheck;
+
+    @Before
+    public void setUp() {
+        TestUtils.stubMethodWithPowerMock();
+    }
+
+    @Test (expected=IllegalArgumentException.class)
+    public void testNullCommitParameter() throws IllegalArgumentException {
+        new LatestCommitCheck(null);
+    }
+
+    @Test (expected=IllegalArgumentException.class)
+    public void testNullCommitPatternParameter() throws IllegalArgumentException {
+        testCommit.setPattern(null);
+        new LatestCommitCheck(testCommit);
+    }
+
+    @Test
+    public void testCheckSimpleRegexMatch() {
+        testCommit.setPattern(Pattern.compile("Test commit"));
+        latestCommitCheck = new LatestCommitCheck(testCommit);
+
+        Assert.assertNull("Cannot match valid regex", latestCommitCheck.check(TestUtils.TEST_PAYLOAD));
+    }
+
+    @Test
+    public void testCheckSimpleRegexNonMatchReturnsExpectedMessage() {
+        testCommit.setPattern(Pattern.compile("can't.*match.*this"));
+        testCommit.setMessage("This is testCommit message.");
+        latestCommitCheck = new LatestCommitCheck(testCommit);
+
+        Assert.assertNotNull("Matched invalid regex", latestCommitCheck.check(TestUtils.TEST_PAYLOAD));
+        Assert.assertEquals("Unexpected message returned", "This is testCommit message.", latestCommitCheck.check(TestUtils.TEST_PAYLOAD));
```

Comment:
if you store the result, you don't need to call this method with the same argument two times